docs/guide/migration.md

```diff
@@ -34,6 +34,10 @@ See the [troubleshooting guide](/guide/troubleshooting.html#vite-cjs-node-api-de
 
 ## General Changes
 
+### `worker.plugins` is now a function
+
+In Vite 4, `worker.plugins` was `(Plugin | Plugin[])[]`. From Vite 5, it needs to be configured as a function `() => (Plugin | Plugin[])[]`. Vite uses parallel rollup builds to bundle workers so new instances of worker plugins are needed for each build.
```

Comment:
```suggestion
In Vite 4, `worker.plugins` accepted an array of plugins (`(Plugin | Plugin[])[]`). From Vite 5, it needs to be configured as a function that returns an array of plugins (`() => (Plugin | Plugin[])[]`). This change is required so parallel worker builds run more consistently and predictably.
```

Added some additional description to entice the reason for the breaking change 😃